src/ConsentRequiredController.js

```diff
@@ -0,0 +1,135 @@
+/*!
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+define([
+  'okta',
+  'shared/util/Util',
+  'util/FormController',
+  'util/FormType',
+  'views/consent/ConsentHeader',
+  'views/consent/ConsentBeacon',
+  'views/consent/ScopeList',
+  'views/consent/ConsentButtonsBar'
+],
+function (Okta, Util, FormController, FormType, ConsentHeader, ConsentBeacon, ScopeList, ConsentButtonsBar) {
+
+  var $ = Okta.$,
+      _ = Okta._;
+
+  return FormController.extend({
+    className: 'consent-required',
+    initialize: function () {
+      this.model.set('expiresAt', this.options.appState.get('transaction').expiresAt);
+      this.model.set('scopes', this.options.appState.get('transaction').scopes);
+    },
+    Model: {
+      props: {
+        expiresAt: ['string', true],
+        scopes: ['array', true]
+      },
+      save: function () {
+        return this.doTransaction(function(transaction) {
+          return transaction.consent({
+            consent: {
+              expiresAt: this.get('expiresAt'),
+              scopes: _.pluck(this.get('scopes'), 'name')
+            }
+          });
+        });
+      },
+      cancel: function () {
+        var self = this;
+        return this.doTransaction(function(transaction) {
+          return transaction.cancel();
+        }).then(function () {
+          Util.redirect(self.settings.get('consentCancelLink'));
+        });
+      }
+    },
+    Form: {
+      noButtonBar: true,
+      formChildren: function () {
+        var transaction = this.options.appState.get('transaction');
+        return [
+          FormType.View({
+            View: new ConsentHeader({
+              orgLogo: this.settings.get('logo'),
+              userFirstName: transaction.user.profile.firstName,
+              userLastName: transaction.user.profile.lastName
+            })
+          }),
+          FormType.View({
+            View: new ConsentBeacon({
+              clientLogo: transaction.target.logo && transaction.target.logo.href,
+            })
+          }),
+          FormType.View({
```

Comment:
I feel like we have too many static views here. Can we get away with just a ConsentHeader, ConsentForm and a ConsentFooter? Where ConsentHeader remains as is, ConsentForm will have the image at the top, the text of 'appname requires consent' and the scopes, and ConsentFooter will have the privacy links

Reply:
This UX is gonna change a lot so I preferred keeping the things very modular